### PR TITLE
Stabilize Python packaging.

### LINKED ISSUE #3: libswiftnav-python build problem
I attempted to build libswiftnav-python with a:
"python setup.py build"
and gcc chokes on a cython generated file:
"swiftnav/dgnss_management.c:5143:3: error: too few arguments to function ‘measure_b_with_external_ambs’"

My Cython version is 0.20.2, and I am running Ubuntu 14.10 Utopic.

John

### LINKED ISSUE #7: libswiftnav-python inconsistent with libswiftnav
On building libswiftnav-python with current libswiftnav, I get:

swiftnav/dgnss_management.c:5099:539: error: too few arguments to function call, expected 6, have 5
  measure_b_with_external_ambs((&(___Pyx_BufPtrCContig1d(__pyx_t_5numpy_double_t *, __pyx_pybuffernd_ref_ecef_.rcbuffer->pybuffer.buf, __pyx_t_18, __pyx_pybuffernd_ref_ecef_.diminfo[0].strides))), __pyx_v_n, (&(__pyx_v_sdiffs[0])), (&(___Pyx_BufPtrCContig1d(__pyx_t_5numpy_double_t _, __pyx_pybuffernd_ambs_.rcbuffer->pybuffer.buf, __pyx_t_20, __pyx_pybuffernd_ambs_.diminfo[0].strides))), (&(___Pyx_BufPtrCContig1d(__pyx_t_5numpy_double_t *, __pyx_pybuffernd_b.rcbuffer->pybuffer.buf, __pyx_t_21, __pyx_pybuffernd_b.diminfo[0].strides))));

### PR DESCRIPTION
**Pull request by mookerji**
Sync and Python distribute issues with libswiftnav-python issues make
it a bit difficult to use the bindings. As part of setting up some
dependent tooling for analysing Piksi's filter state, I'm cleaning up
this project a bit for future use.

Includes:
0. Cleanup of setup.py, docs.
1. Copied over some testing conventions from libsbp.
2. Cleaned up legacy stuff (e.g., float_kf).

Closes #3.
Closes #7.

/cc @fnoble @mfine @cbeighley

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	@echo "  test      to run all tests"
+	@echo
+
+all: c test docs
+
+docs:
+	@echo
+	@echo "Generating Python documentation..."
+	@echo
+	cd $(SWIFTNAV_ROOT)/docs/ && make html
```
Comment (fnoble) [inline Makefile]: consider `make -C`

**Comment by mookerji** [comment]
retest this please

**Comment by mookerji** [comment]
Fuck yeah, Jenkins

**Comment by mookerji** [comment]
won't fix

**Pull request closed without merging**